util/syspolicy/setting/policy_scope.go

```diff
@@ -0,0 +1,187 @@
+	DeviceScope = PolicyScope{kind: DeviceSetting}
+	// CurrentProfileScope indicates a scope containing policies that apply to the
+	// currently active Tailscale profile.
+	CurrentProfileScope = PolicyScope{kind: ProfileSetting}
+	// CurrentUserScope indicates a scope containing policies that apply to the
+	// current user, for whatever that means on the current platform and
+	// in the current application context.
+	CurrentUserScope = PolicyScope{kind: UserSetting}
+)
+
+// PolicyScope is a management scope.
+type PolicyScope struct {
+	kind      Scope
+	userID    string
+	profileID string
+}
+
+// DefaultScope returns the default [PolicyScope] to be used by a program.
+// It returns [DeviceScope], unless explicitly changed with [SetCurrentScope].
+func DefaultScope() PolicyScope {
+	return lazyCurrentScope.Get(func() PolicyScope { return DeviceScope })
+}
+
+// SetCurrentScope attempts to set the specified scope as the default scope.
+// It fails and returns false if called more than once, or if the [DefaultScope]
+// has already been used.
+func SetCurrentScope(scope PolicyScope) bool {
+	return lazyCurrentScope.Set(scope)
+}
+
+// UserScopeOf returns a policy [PolicyScope] of the user with the specified id.
+func UserScopeOf(uid string) PolicyScope {
+	return PolicyScope{kind: UserSetting, userID: uid}
+}
+
+// Kind reports the scope kind of s.
+func (s PolicyScope) Kind() Scope {
+	return s.kind
+}
+
+// IsApplicableSetting reports whether the specified setting applies to
+// and can be retrieved for this scope. Policy settings are applicable
+// to their own scopes as well as more specific scopes. For example,
+// device settings are applicable to device, profile and user scopes,
+// but user settings are only applicable to user scopes.
+// For instance, a menu visibility setting is inherently a user setting
+// and only makes sense in the context of a specific user.
+func (s PolicyScope) IsApplicableSetting(setting *Definition) bool {
+	return setting != nil && setting.Scope() <= s.Kind()
+}
+
+// IsConfigurableSetting reports whether the specified setting can be configured
+// by a policy at this scope. Policy settings are configurable at their own scopes
+// as well as broader scopes. For example, [UserSetting]s are configurable in
+// user, profile, and device scopes, but [DeviceSetting]s are only configurable
+// in the [DeviceScope]. For instance, the InstallUpdates policy setting
+// can only be configured in the device scope, as it controls whether updates
+// will be installed automatically on the device, rather than for specific users.
+func (s PolicyScope) IsConfigurableSetting(setting *Definition) bool {
+	return setting != nil && setting.Scope() >= s.Kind()
+}
+
+// IsWithinOf reports whether policy settings that apply to s2 also apply to s.
+// For example, policy settings that apply to the [DeviceScope] also apply to
+// the [CurrentUserScope].
+func (s PolicyScope) IsWithinOf(s2 PolicyScope) bool {
```

Comment:
Yeah.. Agreed.
That being said, I think `Supercedes` has a slightly different connotation, and `ShouldApplyTo` would make more sense if the receiver was a (setting) `Scope` rather than a `PolicyScope`

But I can probably invert them and rename `IsWithinOf` and `IsStrictlyWithinOf` to `Contains` and `StrictlyContains`.